loadfile/loadfile_test.go

```diff
@@ -20,13 +22,17 @@ func Test_LoadContext(t *testing.T) {
 	testdir := filepath.Join(TestDir, "load-ctx")
 	assert.NoError(t, os.MkdirAll(testdir, os.ModePerm))
 
+	// Randomize the number to prevent collisions when running the test many times with -count
+	//nolint:gosec  // This is not related to security.
+	randSeed := strconv.Itoa(rand.Int())
```

Comment:
Here, `randSeed` is the output of a random number generator. The seed is the number that initializes a random number generator to become the first input into its algorithm.

Also, I think you should create and initialize the seed for the RNG (with a non-zero number), so that its output is reproducibly random, like I did with the [podman filename generation](https://github.com/arcalot/arcaflow-engine-deployer-podman/blob/92bf27cb1f275a8fbe6a790163675e551df531c7/factory.go#L45) (you just don't need to create a random string builder, like [GetRandomString()](https://github.com/arcalot/arcaflow-engine-deployer-podman/blob/92bf27cb1f275a8fbe6a790163675e551df531c7/internal/util/strings.go#L7)).

Reply:
1. Yeah, this isn't a "seed" -- it's a pseudo-random number from the default `Source` with whatever seed it uses.
2. I'm not quite sure how that works across your go funcs: does each have a separate `Source` (which, if each is initialized with the same value, would guarantee conflicts) or if they share a `Source`?
3. If your various test threads aren't sharing a single (thread-safe) `Source`, then each needs to be sure its `Source` is set with a distinct seed.

Reply:
and that access to the rng has to be synchronized to 1 goroutine at a time

Reply:
I was thinking salt, but wrote seed.
I switched to the OS built in functions. That should be plenty reliable for this situation. I took a look at how it's written, and it's a step in reliability.